source/common/quic/envoy_quic_server_stream.cc

```diff
@@ -142,7 +142,7 @@ void EnvoyQuicServerStream::OnInitialHeadersComplete(bool fin, size_t frame_len,
   }
   std::unique_ptr<Http::RequestHeaderMapImpl> headers =
       quicHeadersToEnvoyHeaders<Http::RequestHeaderMapImpl>(header_list);
-  if (!Http::HeaderUtility::authorityIsValid(headers->Host()->value().getStringView())) {
+  if (Http::HeaderUtility::requestHeadersValid(*headers) != absl::nullopt) {
```

Comment:
Nice!

Want to set details here or wait and add them all at once?

Reply:
I will set invalid http header response code in https://github.com/envoyproxy/envoy/pull/15753